src/main/groovy/com/bmuschko/gradle/clover/InstrumentCodeAction.groovy

```diff
@@ -235,7 +236,7 @@ class InstrumentCodeAction implements Action<Task> {
             }
 
             ant.javac(source: getSourceCompatibility(), target: getTargetCompatibility(), encoding: getEncoding(),
-                      executable: getExecutable())
+                      executable: getExecutable(), debug: getDebug())
```

Comment:
I am going to add the flag to groovyc as well and try a unit test as you suggested @bmuschko . I think this is also worth including in the new release.